eng/pipelines/coreclr/perf-non-wasm-jobs.yml

```diff
@@ -45,7 +45,7 @@ jobs:
       platforms:
       - linux_x64
       jobParameters:
-        buildArgs: -s mono+libs+host+packs -c $(_BuildConfig) /p:BuildMonoAOTCrossCompiler=true /p:MonoLibClang="/usr/local/lib/libclang.so.16" /p:AotHostArchitecture=x64 /p:AotHostOS=linux
+        buildArgs: -s mono+libs+host+packs -c $(_BuildConfig) /p:MonoAOTEnableLLVM=true /p:MonoEnableLLVM=true /p:BuildMonoAOTCrossCompiler=true /p:MonoLibClang="/usr/local/lib/libclang.so.16" /p:AotHostArchitecture=x64 /p:AotHostOS=linux
```

Comment:
Without `/p:MonoEnableLLVM=true` x64 job doesn't do AOT-llvm. In contrast, arm64 uses `/p:MonoBundleLLVMOptimizer=true` which we do not set for x64. 

I agree that we should have the same set of flags which in this case would be `/p:MonoAOTEnableLLVM=true /p:MonoBundleLLVMOptimizer=true /p:MonoEnableLLVM=true ...`. However, unifying these depends on if we want to include flags that are expendable for the given platform.

Reply:
If the `MonoEnableLLVM` is set, the `MonoBundleLLVMOptimizer` doesn't need to be explicitly set due to the following condition:
```
    <MonoBundleLLVMOptimizer Condition="'$(MonoEnableLLVM)' == 'true'">true</MonoBundleLLVMOptimizer>
```

Reply:
You are right. I just tested it on osx-arm64 with `/p:MonoAOTEnableLLVM=true` and `/p:MonoEnableLLVM=true` and it was enough to do AOT-llvm. Could you please test it on linux-arm64 so that we are sure that it will be the case on other OSs as well?

Reply:
I have a test run with these changes here: https://dev.azure.com/dnceng/internal/_build/results?buildId=2277114&view=results. The branch I have switched to passing just the AOTEnable and EnableLLVM parameters and the added the mono-aot-cross printout, so the changes will still need to be ported to this PR if successful.

Reply:
The `mono-aot-cross --llvm --version` printout doesn't work for Arm64 Perf job because it gets executed on the x64 machine before being sent to arm64 Helix machine. I will investigate further how we can include this logging so that it gets executed on the correct architecture.